galsim/chromatic.py

```diff
@@ -1359,3 +1770,76 @@ def withScaledFlux(self, flux_ratio):
             return ChromaticAutoCorrelation( self.obj * math.sqrt(flux_ratio), **self.kwargs )
         else:
             return ChromaticObject(self).withScaledFlux(flux_ratio)
+
+class ChromaticOpticalPSF(ChromaticObject):
+    """A subclass of ChromaticObject meant to represent chromatic optical PSFs.
+
+    Chromaticity plays two roles in optical PSFs. First, it determines the diffraction limit, via
+    the wavelength/diameter factor.  Second, aberrations such as defocus, coma, etc. are typically
+    defined in physical distances, but their impact on the PSF depends on their size in units of
+    wavelength.  Other aspects of the optical PSF do not require explicit specification of their
+    chromaticity, e.g., once the obscuration and struts are specified in units of the aperture
+    diameter, their chromatic dependence gets taken care of automatically.
+
+    When using interpolation to speed up image rendering (see ChromaticObject.setupInterpolation()
+    method for details), including ~10-15 samples across any given bandpass should be sufficient.
+    For moderate accuracy, 5 is often sufficient.  Likewise, for high accuracy, `oversample_fac`
+    should be in the range 1.5-2, whereas for moderate accuracy, 1 is often sufficient.  All of
+    these statements assume that aberrations are not very large (typically <~0.25 waves, which is
+    commonly satisfied by space telescopes); if they are larger than that, then more stringent
+    settings are required.
+
+    Also, note that in order to render an image of a real physical object, the ChromaticOpticalPSF
+    should be convolved with some object that has an SED.  Drawing without doing so corresponds to
+    using a flat SED.
+
+    @param   diam          Telescope diameter in meters.
+    @param   aberrations   An array of aberrations, in nanometers.  The size and format of this
+                           array is described in the OpticalPSF docstring.
```

Comment:
>In fact, we could add lambda and diam to OpticalPSF (and maybe even deprecate lam_over_diam which is a bit unwieldy, imo)

I always thought the `lam_over_diam` was a bit weird too, since the natural thing to do is to divide the wavelength by the diameter in consistent length units.  This usually does the wrong thing, of course, since (I assume) most users specify `scale` in arcseconds, or maybe pixels, but not radians.  Having separate `lam` (can't use `lambda`, that's a python reserved word) and `diam` keywords probably makes this even more confusing though, since you would need to specify the wavelength in something like "meters arcseconds" and the diameter in "meters".  We could add a third keyword like `radians_in_one_scale_unit`, I suppose, but then we're starting to make the API unwieldy again.  I like the idea of a unified API, but I'm not sure if it's worth replacing one required argument with three for the non-chromatic `OpticalPSF`.